File: nc103/ReverseString.go
package main

import "fmt"

/**
 * 反转字符串
 * @param str string字符串
 * @return string字符串
 */
func solve(str string) string {
	// write code here

	length := len(str)

	str[0] = 'a'

	return str
}

func main() {
	fmt.Println(solve("012"))
	fmt.Println(solve("1234"))
	fmt.Println(solve("0123456789abcdefghijklmnopqrstuvwxyz"))
}

File: nc103/ReverseString.c
/**
 * 反转字符串
 * @param str string字符串 
 * @return string字符串
 */
char* solve(char* str ) {
    // write code here

    char temp;
    for(int i = 0, j = strlen(str)-1; i < j; i++, j--) {
        temp = str[i];
        str[i] = str[j];
        str[j] = temp;
    }

    return str;
}